dynamiqs/time_qarray.py

```diff
@@ -393,6 +413,12 @@ def __call__(self, t: ScalarLike) -> QArray:
         Returns:
             Qarray evaluated at time $t$.
         """
+        return self._indicator(t) * self._call(t)
```

Comment:
I believe it should be a tiny-bit faster if you return `self._call(t)` or `dq.zero(*self.dims)` depending on the condition, instead of multiplying by 1 or 0. You save both a multiplication and an evaluation of the array if out-of-time-bounds. Plus we won't have surprises regarding dtype promotions due to the multiplication (or stuff like that).